LoopKit/PumpManager.swift

```diff
@@ -50,11 +50,32 @@ public protocol PumpManagerDelegate: PumpManagerStatusObserver {
 
 
 public protocol PumpManager: DeviceManager {
-    // Rounds units to the nearest delivery increment
-    func roundToDeliveryIncrement(units: Double) -> Double
+    /// Rounds a temporary basal rate in U/hr to a rate supported by this pump.
+    ///
+    /// - Parameters:
+    ///   - unitsPerHour: A desired rate of delivery in Units/hr
+    /// - Returns: a supported rate of delivery in Units/hr. The rate returned should not be larger than the passed in rate.
+    func roundToSupportedTemporaryBasalRate(unitsPerHour: Double) -> Double
+
+    /// Rounds a bolus volume in U to a delivery amount supported by this pump.
+    ///
+    /// - Parameters:
+    ///   - units: A desired rate of delivery in U
+    /// - Returns: a supported bolus volume in U. The rate returned should not be larger than the passed in rate.
+    func roundToSupportedBolusVolume(units: Double) -> Double
 
+    // User selectable basal rates
     var supportedBasalRates: [Double] { get }
 
+    // User selectable bolus volumes
+    var supportedBolusVolumes: [Double] { get }
+
+    // Maximum pump allowed bolus volume in U
+    var maximumBolusVolume: Double { get }
+
+    // Maximum pump allowed basal rate in U/Hr
+    var maximumBasalRate: Double { get }
```

Comment:
[nit] I would move this just below `supportedBasalRates` so you have supported/maximum for basal rates, and then supported/maximum for bolus volumes.